src/ILICheck.Web/Validator.cs

```diff
@@ -235,5 +243,31 @@ private IEnumerable<string> ReadGpkgModelNameEntries(string connectionString)
             var reader = command.ExecuteReader();
             while (reader.Read()) yield return reader["modelName"].ToString();
         }
+
+        /// <summary>
+        /// Asynchronously generates a GeoJSON file from the XTF log file.
+        /// </summary>
+        private async Task GenerateGeoJsonAsync()
+        {
+            var xtfLogFile = fileProvider.GetLogFile(LogType.Xtf);
+            if (!fileProvider.Exists(xtfLogFile)) return;
+
+            logger.LogInformation("Generating GeoJSON file from XTF log file <{XtfLogFile}>", xtfLogFile);
+
+            using var reader = fileProvider.OpenText(xtfLogFile);
+            var logResult = XtfLogParser.Parse(reader);
+
+            var featureCollection = GeoJsonHelper.CreateFeatureCollection(logResult);
+            if (featureCollection == null)
+            {
+                logger.LogInformation("No or unknown coordinates found in XTF log file <{XtfLogFile}>. Skipping GeoJSON generation.", xtfLogFile);
+                return;
+            }
+
+            var geoJsonLogFile = Path.ChangeExtension(xtfLogFile, ".geojson");
+
+            using var geoJsonStream = fileProvider.CreateFile(geoJsonLogFile);
+            await JsonSerializer.SerializeAsync(geoJsonStream, featureCollection, jsonOptions.JsonSerializerOptions).ConfigureAwait(false);
```

Comment:
REQ: Die gesamte Logik in einen `try..catch`-Block packen, so dass wir bei unerwarteten Fehlern nicht die gesamte Validierung abbrechen, wenn das GeoJSON-Log nicht erstellt werden konnte.